src/test/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceMultiSourceConfigTest.java

```diff
@@ -0,0 +1,138 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce.plugin.source.batch;
+
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.cdap.etl.api.validation.ValidationFailure;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link SalesforceMultiSourceConfig}.
+ */
+public class SalesforceMultiSourceConfigTest {
+
+    @Test
+    public void testGetWhiteList() {
+        String whiteList = "Account,Contact";
+        SalesforceMultiSourceConfig config = new SalesforceMultiSourceConfigBuilder()
+                .setWhiteList(whiteList)
+                .build();
+
+        Set<String> sObjects = config.getWhiteList();
+        Assert.assertNotNull(sObjects);
+        Assert.assertEquals(whiteList, String.join(",", sObjects));
+
+    }
+
+    @Test
+    public void testGetBlackList() {
+        String blackList = "Lead,Contact";
+        SalesforceMultiSourceConfig config = new SalesforceMultiSourceConfigBuilder()
+                .setBlackList(blackList)
+                .build();
+
+        Set<String> sObjects = config.getBlackList();
+        Assert.assertNotNull(sObjects);
+        Assert.assertEquals(blackList, String.join(",", sObjects));
+
+    }
+
+    @Test
+    public void testGetSObjectNameField() {
+        String sObjectFieldName = "Account";
+        SalesforceMultiSourceConfig config = new SalesforceMultiSourceConfigBuilder()
+                .setsObjectNameField(sObjectFieldName)
+                .build();
+
+        String sObject = config.getSObjectNameField();
+        Assert.assertNotNull(sObject);
+        Assert.assertEquals(sObjectFieldName, sObject);
+    }
+
+    @Test
+    public void testValidate() {
+        SalesforceMultiSourceConfig config = new SalesforceMultiSourceConfigBuilder().setsObjectNameField("Account")
+                .setDuration("kkk").
+                build();
+        MockFailureCollector collector = new MockFailureCollector();
+        SalesforceMultiSourceConfig mock = Mockito.spy(config);
+        when(mock.canAttemptToEstablishConnection()).thenReturn(false);
+        ValidationFailure failure;
+        try {
+            mock.validate(collector);
+            Assert.assertEquals(1, collector.getValidationFailures().size());
+            failure = collector.getValidationFailures().get(0);
+        } catch (ValidationException e) {
+            Assert.assertEquals(1, e.getFailures().size());
+            failure = e.getFailures().get(0);
+        }
+
```

Comment:
fail if exception is not thrown

Reply:
On test re-run, it was observed that no exception could be thrown. So removed the `try-catch` block.